### PR TITLE
input valid check

### PR DESCRIPTION
**Pull request by JiaYiChe**
Built input check

### CODE REVIEW

File: project/app/src/main/java/com/example/cmput301w21t25/activities_experiments/CreateExperimentActivity.java [hunk 1]
```diff
@@ -88,37 +89,47 @@ public void onClick(View v) {
```
Comment (dkrammer) [inline project/app/src/main/java/com/example/cmput301w21t25/activities_experiments/CreateExperimentActivity.java:91]: This is a pretty huge method, we should try to refactor this if we have time (not high priority right now though)

File: project/app/src/main/java/com/example/cmput301w21t25/activities_experiments/CreateExperimentActivity.java [hunk 2]
```diff
@@ -177,4 +188,44 @@ public void onRadioButtonClicked(View view) {
...
+        }
+        return checkIfValid;
+    }
+
+    /**
+     * This method is to check whether the checkbox got selected
+     * return true if selected, false for not
+     * @return
+     */
+    public boolean checkBoxSelecte(){
```
Comment (dkrammer) [inline project/app/src/main/java/com/example/cmput301w21t25/activities_experiments/CreateExperimentActivity.java]: I recently added in the functionality for the "Subscribe to experiment?" checkbox as well, I'm not sure if it's necessary but could be worth adding that the check for that in here now that it actually does something? It's an optional input though, so I'm not sure it even needs to be checked.

File: project/app/src/main/java/com/example/cmput301w21t25/activities_user/MyUserProfileActivity.java [hunk 3]
```diff
@@ -138,14 +138,19 @@ public void updateButton(View view) {
...
         String email = newEmail.getText().toString();
-        UserManager userManager = new UserManager();
 
-        if (!name.equals("") && !email.equals("")) {
-            userManager.FB_isUnique(name, userID, email, this,  "update");
-        }
-        else {
-            //something is null
-            Toast.makeText(MyUserProfileActivity.this, "Don't leave fields empty!", Toast.LENGTH_SHORT).show();
+        if(name.length()>0 && email.length()>0){
```
Comment (dkrammer) [inline project/app/src/main/java/com/example/cmput301w21t25/activities_user/MyUserProfileActivity.java]: It looks like the FB_isUnique() call was removed, which we need in here somewhere.

**Review by dkrammer — CHANGES_REQUESTED** [review]
I left some comments on things I thought could or should be changed at some point, I'll leave it to Curtis to give a more thorough look at the database stuff since I don't know as well as he would. Only necessary change is that the FB_isUnique method call in MyUserProfileActivity.java was removed, so that needs to be added back in. The rest of my comments are just notes.

**Review by dkrammer — APPROVED** [review]
Looks good!

File: project/app/src/main/java/com/example/cmput301w21t25/activities_experiments/CreateExperimentActivity.java (hunk 2 above)
Comment (curtiskennedy) [reply project/app/src/main/java/com/example/cmput301w21t25/activities_experiments/CreateExperimentActivity.java]: I think all the checkboxes are optional? This method may not be needed then.

File: project/app/src/main/java/com/example/cmput301w21t25/activities_experiments/CreateExperimentActivity.java [hunk 4]
```diff
@@ -88,37 +89,47 @@ public void onClick(View v) {
...
+                    experimentKeywords = new ArrayList<String>();
+                    String keywords = experimentTags.getText().toString();
+                    experimentKeywords = parseKeywords(keywords);
+
+
+                    Log.d("description", description);
+                    Log.d("name", name);
+                    Log.d("keywords", experimentKeywords.toString());
+                    Toast toast = Toast.makeText(getApplicationContext(), "Please don't skip enter informations", Toast.LENGTH_LONG);
+                    if(!checkEnterValid(description,name,experimentKeywords) || type==null || !checkBoxSelecte()){
```
Comment (curtiskennedy) [inline project/app/src/main/java/com/example/cmput301w21t25/activities_experiments/CreateExperimentActivity.java]: see below regarding the checkboxes

File: project/app/src/main/java/com/example/cmput301w21t25/activities_user/MyUserProfileActivity.java [hunk 5]
```diff
@@ -138,14 +138,19 @@ public void updateButton(View view) {
         EditText newEmail = findViewById(R.id.updateEmail);
         String name = newName.getText().toString();
         String email = newEmail.getText().toString();
-        UserManager userManager = new UserManager();
 
-        if (!name.equals("") && !email.equals("")) {
```
Comment (curtiskennedy) [inline project/app/src/main/java/com/example/cmput301w21t25/activities_user/MyUserProfileActivity.java]: Input validation already exists here so I think you can leave this file as-is.

File: project/app/src/main/java/com/example/cmput301w21t25/activities_experiments/CreateExperimentActivity.java (hunk 4 above)
Comment (curtiskennedy) [reply project/app/src/main/java/com/example/cmput301w21t25/activities_experiments/CreateExperimentActivity.java]:
```suggestion
                    if(!checkEnterValid(description,name,experimentKeywords) || type==null){
```

**Review by curtiskennedy — CHANGES_REQUESTED** [review]
Nice work, thanks for taking care of this! My only suggestions are for the CreateExperimentActivity (the checkBoxSelect() method may not be needed since I believe the checkboxes are optional) as well as the MyUserProfileActivity (like the GenerateUserActivity, it already has input validation, so you can leave the file as it was before)

**Review by curtiskennedy — APPROVED** [review]
Looks good!

**Pull request merged**